pkg/sql/upsert.go

```diff
@@ -126,7 +126,7 @@ func (p *planner) newUpsertNode(
 		// Determine which columns are updated by the RHS of INSERT
 		// ... ON CONFLICT DO UPDATE, or the non-PK columns in an
 		// UPSERT.
-		names, err := p.namesForExprs(updateExprs)
+		names, _, err := p.namesForExprs(ctx, updateExprs)
```

Comment:
that's not the end of the story unfortunately... Either:

- you must the resulting setExprs in the same way UPDATE does. This is "interesting" because it requires the same "slot" machinery that was created for UPDATE by Justin.

- or, you can throw the towel and return a "not supported" error. I'd be fine with that -- we never supported subqueries in that place before.

I think the same problem exists with tuple assignments too (eg `ON CONFLICT DO UPDATE (a,b) = (c+d,e+f)` since the slot solution was intended to solve that problem too)